snowflake/exec.go

```diff
@@ -0,0 +1,13 @@
+package snowflake
+
+import (
+	"database/sql"
+
+	"github.com/sirupsen/logrus"
+)
+
+func ExecNoRows(db *sql.DB, query string) (sql.Result, error) {
+	logrus.Debug("[DEBUG] exec stmt ", query)
+
+	return db.Exec(query)
+}
```

Comment:
I think I'm missing something, what makes this a "no rows" operation?

Reply:
Oh, if you ctrl+click into the function, you see this:
```
// Exec executes a query without returning any rows.
// The args are for any placeholder parameters in the query.
func (db *DB) Exec(query string, args ...interface{}) (Result, error) {
	return db.ExecContext(context.Background(), query, args...)
}
```
then here:
```
// ExecContext executes a query without returning any rows.
// The args are for any placeholder parameters in the query.
func (db *DB) ExecContext(ctx context.Context, query string, args ...interface{}) (Result, error) {
	var res Result
	var err error
	for i := 0; i < maxBadConnRetries; i++ {
		res, err = db.exec(ctx, query, args, cachedOrNewConn)
		if err != driver.ErrBadConn {
			break
		}
	}
	if err == driver.ErrBadConn {
		return db.exec(ctx, query, args, alwaysNewConn)
	}
	return res, err
}
```
Anything deeper than this is a mystery to me, but the comments told me that there are no rows.

Reply:
We can add statements with rows though. [Link](https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/pkg/snowflake/exec.go#L34-L49)